spec/models/thermometer_widget_spec.rb

```diff
@@ -1,41 +1,31 @@
 describe ThermometerWidget do
 
-  let(:content) { {
-    goal: 10000,
-    count: 500,
-    autoincrement: true
-  } }
-  let(:string_content) { {
-    goal: "12345",
-    count: "45",
-    autoincrement: "0"
-  } }
-  let(:desired_content) { {
-    goal: 12345,
-    count: 45,
-    autoincrement: false
-  } }
-  let(:params) { { page_display_order: 1, content: content } }
-  let(:widget) { ThermometerWidget.create!(params) }
+  let(:params) do
+    {
+      goal: "12345",
+      count: "45",
+      autoincrement: "0",
+      page_display_order: 1
+    }
+  end
 
-  subject { widget }
-  it { should be_valid }
+  before do
+    ThermometerWidget.create!(params)
+  end
 
-  describe 'content types' do
+  subject { Widget.first }
 
-    before :each do
-      params[:content] = string_content
-    end
+  it { should be_valid }
 
-    it "should be able to cast the values to the correct type on create" do
-      expect{ThermometerWidget.create!(params)}.to change{ Widget.count }.by 1
-      expect(ThermometerWidget.last.content).to eq desired_content.with_indifferent_access
+  describe 'content types' do
+    it 'casts values to the correct type on create' do
+      expect(subject.content['count']).to eq(45)
+      expect(subject.content['autoincrement']).to be false
     end
 
-    it "should be able to cast the values to the correct type on update" do
-      w = widget # reference widget so lazy let is called
-      expect{widget.update_attributes!(params)}.to change{ Widget.count }.by 0
-      expect(widget.reload.content).to eq desired_content.with_indifferent_access
+    it "casts values to the correct type on update" do
+      subject.update_attributes!(autoincrement: '1')
+      expect(subject.content['autoincrement']).to be true
     end
```

Comment:
@NealJMD I've updated this spec to show typecast values are being persisted, so `true` instead of `1`, etc.